### PR TITLE
Change app menu to white background with dark icons

### PR DESCRIPTION
**Pull request by juliusknorr**
This PR changes the app menu to white background with dark icons, as proposed in #455 

2016-07-28-135630_957x282_scrot

Chrome, Firefox and Safari seem to work fine, could someone please verify that the inverting of icons works with IE/Edge?

Please review @jancborchardt @nextcloud/designers @Bugsbane @icewind1991 @williambargent

### CODE REVIEW

**Comment by williambargent** [comment]
The icons don't show in IE and only show half in Edge.

capture

File: core/css/header.css [hunk 1]
```diff
@@ -252,8 +257,8 @@
 	height: 32px;
 }
 #navigation .app-loading .app-icon {
-	-ms-filter: "progid:DXImageTransform.Microsoft.Alpha(Opacity=10)";
-	opacity: .1;
+	-ms-filter: "progid:DXImageTransform.Microsoft.Alpha(Opacity=0)";
```
Comment (nickvergessen) [inline core/css/header.css]: you mean 90?

File: core/css/header.css [hunk 2]
```diff
@@ -252,8 +257,8 @@
...
+	opacity: 0;
```
Comment (nickvergessen) [inline core/css/header.css]: and 0.9 ?

File: core/css/header.css (hunk 1 above)
Comment (juliushaertl) [inline core/css/header.css]: No, as the icon should not be visible anymore, when the loading indicator is shown.

**Comment by juliushaertl** [comment]
> The icons don't show in IE and only show half in Edge.

Ok, will try to fix that later today.

**Comment by jancborchardt** [comment]
@juliushaertl awesome! Only thing from my side is that it looks a bit harsh – the icons could be lighter. Here it is with active changed from .8 to .6 and the others from .6 to .4:
Before & after:
capture du 2016-07-28 16-30-09
capture du 2016-07-28 16-30-48

(We should also do that in the users menu on the top right then :)

@juliushaertl if it’s ok I can add those commits, or you can feel free to do it. Don’t want to mess with your branch while you’re working on it. ;)

**Comment by juliushaertl** [comment]
@jancborchardt Right, that looks far nicer. Feel free to commit, otherwise I could also do that later this evening.

File: core/css/header.css [hunk 3]
```diff
@@ -204,28 +204,33 @@
...
+		-ms-filter: "progid:DXImageTransform.Microsoft.Alpha(Opacity=75)";
+		opacity: .75;
 	}
 
 #navigation .app-icon {
+  -webkit-filter: invert(100%);
+  -moz-filter: invert(100%);
+  -o-filter: invert(100%);
+  -ms-filter: "progid:DXImageTransform.Microsoft.Invert(100%)";
+  filter: invert(100%);
```
Comment (jancborchardt) [inline core/css/header.css]: @juliushaertl please always use tabs ;) according to our coding style guidelines and what’s used elsewhere in the file.

**Comment by jancborchardt** [comment]
@juliushaertl made the changes. Also, you need to use tabs instead of spaces ;D

**Comment by juliushaertl** [comment]
Pushed a new commit that now also supports IE 9+, Edge.

Please review again @jancborchardt @nickvergessen @williambargent

**Comment by schiessle** [comment]
This looks really nice, great work!

I just have a small issue. It seems like the "App" menu entry is handled differently then all the apps. The icon of the current active app is darker and the other icons change to the dark color on hover, expect the "apps" icon (there is also a hover effect but not as strong as for the other apps). If I go to the app settings the apps icon also doesn't change to the darker color by default.

Tested with Chromium 52

**Comment by jancborchardt** [comment]
@schiessle this is by design because the »Apps« entry is different from the others. It’s styled a bit more transparent accordingly to not be confused with an app.

**Comment by jancborchardt** [comment]
@williambargent mind retesting this? :)

**Comment by schiessle** [comment]
> @schiessle this is by design because the »Apps« entry is different from the others. It’s styled a bit more transparent accordingly to not be confused with an app.

Hm, but if they are not selected all look the same, including the app menu. I consider it rather confusing that the hover effects behaves different on some elements on the same menu and especially confusing that I don't have a visual feedback that the current page is the app menu.

**Comment by juliushaertl** [comment]
How about keeping the Apps icon a bit lighter but changing the active state to the same grey as the other icons? @jancborchardt @schiessle

**Comment by williambargent** [comment]
Awesome, its almost working. When you hover over 'add apps' in Edge the colour doesn't change but does in IE. Also when your on the 'add apps' page it doesn't show that it has been selected.

**Comment by schiessle** [comment]
@williambargent That's what I already mentioned here: https://github.com/nextcloud/server/pull/627#issuecomment-236118554

imho the app menu should behave the same as any other entry with respect to the hover effect and the look if it is selected. Might be OK if it is a bit lighter if it is not selected to highlight the real apps a bit. But hover effect and selected color should be exactly the same.

**Comment by jancborchardt** [comment]
> How about keeping the Apps icon a bit lighter but changing the active state to the same grey as the other icons?

@juliushaertl sounds perfect! Fixed that :)

**Comment by juliushaertl** [comment]
@jancborchardt looks good :)

**Comment by schiessle** [comment]
Hm, I still see no visible hint that I'm at the apps page:

menu

**Comment by jancborchardt** [comment]
@schiessle try hard-refreshing (Ctrl-Shift-R) or incognito mode:
capture du 2016-07-29 18-56-35
_(I know, nice theme right? ;D)_

**Comment by mar1u5** [comment]
Works for me :+1:

**Comment by schiessle** [comment]
> try hard-refreshing (Ctrl-Shift-R) or incognito mode:

Did all that but I should have done a git pull before testing it again :wink: 

Looks good now :+1:

**Comment by schiessle** [comment]
I would suggest to backport this for Nextcloud 10, objections?

**Comment by mar1u5** [comment]
Definetly. :+1: What do you think? @karlitschek

**Comment by karlitschek** [comment]
looks good. please backport 👍

**Comment by jancborchardt** [comment]
@juliushaertl do you want to open a backport PR of your awesome work to stable10? :) Probably directly with https://github.com/nextcloud/server/pull/664 on top.

**Comment by juliushaertl** [comment]
@jancborchardt just did at #665

**Pull request merged**